atomicapp/utils.py

```diff
@@ -20,6 +20,20 @@
 def isTrue(val):
     return str(val).lower() in true_values
 
+def addShelveKV(dbname, key, val):
+    with closing(shelve.open(dbname, writeback = True)) as s:
+        s[key] = val
+
+def appendShelveKV(dbname, key, val):
+    with closing(shelve.open(dbname, writeback = True)) as s:
+        temp = s[key]
+        temp.append(val)
```

Comment:
ah ok i have  writeback = True ,so you dont need it. writeback = True ; will cache the data and rewrite it.